articles/validating-xml-using-xsd/index.md

```diff
@@ -0,0 +1,262 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /engineering-education/validating-xml-using-xsd/
+title: Validating XML using XSD
+description: A tutorial on validating XML using XSD
+author: srishilesh-p-s
+date: 2020-11-15T00:00:00-12:00
+topics: []
+excerpt_separator: <!--more-->
+images:
+  - url: /engineering-education/validating-xml-using-xsd/hero.jpg
+    alt: Validating XML using XSD
+---
+
+In this article, we will understand how validation of XML is done using XSD. By the end of this article, you will get an overview of various techniques for validating XML tags and attributes using XSD. You will also be learning step by step guide for validation. It is highly recommended to go through my [previous article](), to understand the basics of XML and its importance of validation.
+
+<!--more-->
+
+### Table of contents
+- [Introduction](#introduction)
+- [What is XSD?](#what-is-xsd)
+- [Step by step guide for validation](#step-by-step-guide-for-validation)
+- [Conclusion](#conclusion)
+- [Further reading](#further-reading)
+
+### Introduction
+In our [previous article](), we understood various data serialization techniques like XML, the importance of validating XML, and also validated an XML schema using DTD.
+
+[Document Type Definition (DTD)](https://en.wikipedia.org/wiki/Document_type_definition) helped us in validating the XML schema by parsing the structure of the XML document. But, we can't rely entirely on validation using DTD, since there are high chances of faulty data or data types mismatch in values. This lead to the improvement in the validation technique, where apart from parsing the XML structure, we also understand the semantics of the schema, using XSD.
```

Comment:
> This lead to the improvement in the validation technique, where apart from parsing the XML structure, we also understand the semantics of the schema, using XSD.

This sentence is a little too long and lacks readability. Could you please break it down into 2-3 sentences?